nodes/move-object/move-object.js

```diff
@@ -0,0 +1,153 @@
+module.exports = function (RED) {
+  "use strict";
+  const { S3 } = require("@aws-sdk/client-s3");
+
+  // Move Object
+  function S3MoveObject(n) {
+    RED.nodes.createNode(this, n); // Getting options for the current node
+    this.conf = RED.nodes.getNode(n.conf); // Getting configuration
+    var node = this; // Referencing the current node
+    var config = this.conf ? this.conf : null; // Cheking if the conf is valid
+
+    // If there is no conifg
+    if (!config) {
+      node.warn(RED._("Missing S3 Client Configuration!"));
+      return;
+    }
+
+    this.on("input", async function (msg, send, done) {
+      let bucket = n.bucket !== "" ? n.bucket : null; // Bucket info
+      let key = n.key !== "" ? n.key : null; // Object key
+
+      let sourcekey = n.sourcekey !== "" ? n.sourcekey : null;
+      let sourcebucket = n.sourcebucket !== "" ? n.sourcebucket : null;
+
+      // Checking for correct properties input
+      if (!bucket) {
+        bucket = msg.bucket ? msg.bucket : null;
+        if (!bucket) {
+          node.error("No bucket provided!");
+          return;
+        }
+      }
+
+      if (!key) {
+        key = msg.key ? msg.key : null;
+        if (!key) {
+          node.error("No object key provided!");
+          return;
+        }
+      }
+
+      if (!sourcebucket) {
+        sourcebucket = msg.sourcebucket ? msg.sourcebucket : null;
+        if (!sourcebucket) {
+          node.error("No sourcebucket provided!");
+          return;
+        }
+      }
+
+      if (!sourcekey) {
+        sourcekey = msg.sourcekey ? msg.sourcekey : null;
+        if (!sourcekey) {
+          node.error("No sourcekey provided!");
+          return;
+        }
+      }
+
+      // S3 client init
+      let s3Client = null;
+
+      try {
+        // Creating S3 client
+        s3Client = new S3({
+          endpoint: config.endpoint,
+          forcePathStyle: config.forcepathstyle,
+          region: config.region,
+          credentials: {
+            accessKeyId: config.credentials.accesskeyid,
+            secretAccessKey: config.credentials.secretaccesskey,
+          },
+        });
+
+        // Uploading
+        node.status({
+          fill: "blue",
+          shape: "dot",
+          text: "Uploading",
```

Comment:
The status should represent correctly what is the current state of the node, so in this case I would suggest updating the text property to `Moving` or similar